homeassistant/components/ring/__init__.py

```diff
@@ -2,34 +2,39 @@
 
 from __future__ import annotations
 
+from dataclasses import dataclass
 from functools import partial
 import logging
+from typing import Any, cast
 
-from ring_doorbell import Auth, Ring
+from ring_doorbell import Auth, Ring, RingDevices
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import APPLICATION_NAME, CONF_TOKEN, __version__
 from homeassistant.core import HomeAssistant, ServiceCall, callback
 from homeassistant.helpers import device_registry as dr, entity_registry as er
 from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
 
-from .const import (
-    DOMAIN,
-    PLATFORMS,
-    RING_API,
-    RING_DEVICES,
-    RING_DEVICES_COORDINATOR,
-    RING_NOTIFICATIONS_COORDINATOR,
-)
+from .const import DOMAIN, PLATFORMS
 from .coordinator import RingDataCoordinator, RingNotificationsCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class RingData:
+    """Class to support type hinting of ring data collection."""
+
+    ring_api: Ring
+    ring_devices: RingDevices
+    ring_devices_coordinator: RingDataCoordinator
+    ring_notifications_coordinator: RingNotificationsCoordinator
```

Comment:
```suggestion
    api: Ring
    devices: RingDevices
    devices_coordinator: RingDataCoordinator
    notifications_coordinator: RingNotificationsCoordinator
```
Just so it's a bit more concise, I'd suggest to drop the `ring_` prefix here. The whole data structure references ring data.

Reply:
Makes sense, that's done